### PR TITLE
Don't load current_user progress when `?solution=true`

### PR DESCRIPTION
**Pull request by joshlory**

### CODE REVIEW

File: dashboard/app/assets/javascripts/client_state.js [hunk 1]
```diff
@@ -36,6 +36,22 @@ dashboard.clientState.reset = function() {
   sessionStorage.clear();
 };
 
+dashboard.clientState.queryParams = function (name) {
```
Comment (islemaster) [inline dashboard/app/assets/javascripts/client_state.js:44]: This method has a somewhat complicated contract (with the `name` argument being optional, returning an Object if it's omitted, a string if it's not).  Can we add documentation?

File: shared/js/loadApp.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ module.exports = function (callback) {
     }
   };
 
-  if (!appOptions.channel) {
+  if (!appOptions.channel && dashboard.clientState.queryParams('solution') !== 'true') {
```
Comment (islemaster) [inline shared/js/loadApp.js]: Nit: Would be nice to extract this query param check into a descriptive helper (e.g. `... && !viewingSolution()`).

**Comment by islemaster** [comment]
Couple of nits, change looks safe.  Thanks for fixing!  LGTM

File: shared/js/loadApp.js (hunk 2 above)
Comment (islemaster) [inline shared/js/loadApp.js]: Descriptive local variable would work too; I guess a function might be overkill.

**Comment by philbogle** [comment]
LGTM, thanks for fixing.

File: shared/js/loadApp.js [hunk 3]
```diff
@@ -29,7 +29,9 @@ module.exports = function (callback) {
     }
   };
 
-  if (!appOptions.channel) {
+  var isViewingSolution = (dashboard.clientState.queryParams('solution') === 'true');
```
Comment (laurelfan) [inline shared/js/loadApp.js:32]: the rails code tests for the presence of params['solution'] not that it === 'true' so this is slightly different from the condition there...

**Pull request merged**